internal/pkg/composable/providers/kubernetes/hints_test.go

```diff
@@ -367,6 +367,145 @@ func TestGenerateHintsMappingWithLogStream(t *testing.T) {
 	assert.Equal(t, expected, hintsMapping)
 }
 
+func TestGenerateHintsMappingWithDefaultsandTypo(t *testing.T) {
+	logger := getLogger()
+	pod := &kubernetes.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "testpod",
+			UID:       types.UID(uid),
+			Namespace: "testns",
+			Labels: map[string]string{
+				"foo": "bar",
+			},
+			Annotations: map[string]string{
+				"app": "production",
+			},
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		Spec: kubernetes.PodSpec{
+			NodeName: "testnode",
+		},
+		Status: kubernetes.PodStatus{PodIP: "127.0.0.5"},
+	}
+
+	mapping := map[string]interface{}{
+		"namespace": pod.GetNamespace(),
+		"pod": mapstr.M{
+			"uid":  string(pod.GetUID()),
+			"name": pod.GetName(),
+			"ip":   pod.Status.PodIP,
+		},
+		"namespace_annotations": mapstr.M{
+			"nsa": "nsb",
+		},
+		"labels": mapstr.M{
+			"foo": "bar",
+		},
+		"annotations": mapstr.M{
+			"app": "production",
+		},
+	}
+	hints := mapstr.M{
+		"hints": mapstr.M{
+			"host":        "${kubernetes.pod.ip}:6379",
+			"package":     "redis",
+			"metricspath": "/metrics", // on purpose we have introduced a typo
+			"timeout":     "42s",
+			"period":      "42s",
+		},
+	}
+
+	expected := mapstr.M{
+		"redis": mapstr.M{
+			"enabled": true,
+			"host":    "127.0.0.5:6379",
+			"timeout": "42s",
+			"period":  "42s",
+		},
+	}
+
+	hintsMapping := GenerateHintsMapping(hints, mapping, logger, "")
+
+	assert.Equal(t, expected, hintsMapping)
+}
+
+func TestGenerateHintsMappingWithInfoandTypo(t *testing.T) {
+	logger := getLogger()
+	pod := &kubernetes.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "testpod",
+			UID:       types.UID(uid),
+			Namespace: "testns",
+			Labels: map[string]string{
+				"foo":        "bar",
+				"with-dash":  "dash-value",
```

Comment:
can it be removed as well? the same in `pod_test`

Reply:
Done thanks!

Reply:
Done in [62d1b2d](https://github.com/elastic/elastic-agent/pull/4360/commits/62d1b2d7af52bfe09406cff957b6be838b147d1c)